src/passes/OptimizeInstructions.cpp

```diff
@@ -537,6 +536,11 @@ struct OptimizeInstructions
               (op = makeUnsignedBinaryOp(binary->op)) != InvalidBinary &&
               Bits::getMaxBits(binary->left, this) <= 31) {
             binary->op = op;
+          } else if (c < 0 && c > std::numeric_limits<int32_t>::min() &&
+                     binary->op == DivUInt32) {
+            // u32(x) / -C   ==>   u32(x) >= -C
```

Comment:
This might be more obviously correct if framed this way, since it doesn't mix negative numbers with unsigned operations:

```suggestion
            // u32(x) / C   ==>   u32(x) >= C  iff C >= 2^31
```

wdyt?